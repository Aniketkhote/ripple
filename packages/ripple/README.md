<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/trueadm/ripple/master/assets/ripple-dark.png">
  <img src="https://raw.githubusercontent.com/trueadm/ripple/master/assets/ripple-light.png" alt="Ripple - the elegant UI framework for the web" />
</picture>

# What is Ripple?

> Currently, this project is still in early development, and should not be used in production.

Ripple is a TypeScript UI framework for the web. To find out more, view [Ripple's Github README](https://github.com/trueadm/ripple).
