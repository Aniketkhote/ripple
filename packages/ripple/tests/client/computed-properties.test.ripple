import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { mount, flushSync, track, TrackedArray, TrackedMap, effect } from 'ripple';

describe('computed tracked properties', () => {
	let container;

	function render(component) {
		mount(component, {
			target: container
		});
	}

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

  it('should update a property using assignment', () => {
    component App() {
      let obj = {
        [0]: track(0),
      };

      <div>{obj.@[0]}</div>

      <button onClick={() => { obj.@[0] += 1 }}>{"Increment"}</button>
    }

    render(App);
    expect(container).toMatchSnapshot();

    const button = container.querySelector('button');
    button.click();
    flushSync();

    expect(container).toMatchSnapshot();
  });

  it('should update a property using update expressions', () => {
    component App() {
      let obj = {
        [0]: track(0),
      };

      <div>{obj.@[0]}</div>

      <button onClick={() => { obj.@[0]++ }}>{"Increment"}</button>
    }

    render(App);
    expect(container).toMatchSnapshot();

    const button = container.querySelector('button');
    button.click();
    flushSync();

    expect(container).toMatchSnapshot();
  });
});