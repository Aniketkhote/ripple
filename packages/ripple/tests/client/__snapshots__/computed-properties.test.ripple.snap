// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`computed tracked properties > should update a property using assignment 1`] = `
<div>
  <div>
    0
  </div>
  <button>
    Increment
  </button>
  
</div>
`;

exports[`computed tracked properties > should update a property using assignment 2`] = `
<div>
  <div>
    1
  </div>
  <button>
    Increment
  </button>
  
</div>
`;

exports[`computed tracked properties > should update a property using update expressions 1`] = `
<div>
  <div>
    0
  </div>
  <button>
    Increment
  </button>
  
</div>
`;

exports[`computed tracked properties > should update a property using update expressions 2`] = `
<div>
  <div>
    1
  </div>
  <button>
    Increment
  </button>
  
</div>
`;
